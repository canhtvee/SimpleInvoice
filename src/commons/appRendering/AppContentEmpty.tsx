import React from 'react';
import {StyleProp, TextStyle, View, ViewStyle} from 'react-native';

import {Sizes, Strings} from '../../utils';

import {AppText} from '../appText';

export interface AppContentEmptyProps {
  title?: string;
  titleStyle?: StyleProp<TextStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  onPress?: () => void;
}

export function AppContentEmpty({
  titleStyle,
  containerStyle,
}: AppContentEmptyProps) {
  return (
    <View
      style={[
        {
          width: '100%',
          alignItems: 'center',
          marginTop: Sizes.wpx(60),
        },
        containerStyle,
      ]}>
      <AppText
        style={[
          {
            paddingHorizontal: Sizes.padding,
            textAlign: 'center',
          },
          titleStyle,
        ]}>
        {Strings.contentEmpty}
      </AppText>
    </View>
  );
}
