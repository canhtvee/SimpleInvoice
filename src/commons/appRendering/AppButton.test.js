import React from 'react';

import {cleanup} from '@testing-library/react-native';
import {create} from 'react-test-renderer';

import {AppContentEmpty} from './AppContentEmpty';

/**
 * Run test command
 * yarn test -u -t="AppContentEmpty"
 */

afterEach(cleanup);

describe('AppContentEmpty', () => {
  test('should render correctly', () => {
    const tree = create(<AppContentEmpty />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
