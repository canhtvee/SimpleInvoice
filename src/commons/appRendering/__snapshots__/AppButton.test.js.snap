// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContentEmpty should render correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "marginTop": 118.5,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 27.5,
        },
        [
          {
            "paddingHorizontal": 23.5,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    Data is not available
  </Text>
</View>
`;
